source/reference/connection-string.txt

```diff
@@ -51,6 +51,8 @@ The components of this string are:
        specific database using these credentials after connecting to
        the :program:`mongod` instance.
 
+       If username or password contains an at-sign ("@") or colon (":") then the username and/or password must be URL encoded. At-sign ("@") is replaced with "%40" (3 characters), while colon (":") is replaced with "%3A" (3 characters).
+
```

Comment:
Yes, I agree, it looks like all the drivers should be tested for this condition and bug reports raised for anything outside the spec. I think for the moment we use our specification as the definitive guide to what the common behavior should be (i.e escape @ and : symbols).

It would be nice to have the spec define what should happen if other escape sequences are present since people using Node (in particular, but also other languages) are likely to use standard libs to apply HTTP escaping, which will escape all manner of characters, not just 'at' and colon.